src/KafkaFlow.OpenTelemetry/OpenTelemetryProducerEventsHandler.cs

```diff
@@ -48,7 +48,7 @@ public static Task OnProducerStarted(IMessageContext context)
                 // Inject the ActivityContext into the message headers to propagate trace context to the receiving service.
                 Propagator.Inject(new PropagationContext(contextToInject, Baggage.Current), context, InjectTraceContextIntoBasicProperties);
 
-                ActivitySourceAccessor.SetGenericTags(activity);
+                ActivitySourceAccessor.SetGenericTags(activity, context?.ProducerContext.Brokers);
```

Comment:
It can pass a null value, is SetGenericTags prepared for that?

Reply:
Yes. SetGenericTags only adds tags to activity object. `System.Diagnostics.Activity` has a fail-safe that, when the tag value is null, the tag is not set.